src/KeyHandlerTest.cpp

```diff
@@ -315,23 +315,24 @@ TEST_F(KeyHandlerTest,
   auto endState = handleKeySequence(keys);
   auto inputtingState = dynamic_cast<InputStates::Inputting*>(endState.get());
   ASSERT_TRUE(inputtingState != nullptr);
-  ASSERT_EQ(inputtingState->composingBuffer, "以");
-  ASSERT_EQ(inputtingState->cursorIndex, strlen("以"));
+  ASSERT_EQ(inputtingState->composingBuffer, "一ˇ");
+  ASSERT_EQ(inputtingState->cursorIndex, strlen("一ˇ"));
 }
 
 TEST_F(KeyHandlerTest, ToneMarkThenNonToneComponentOnlyComposesWithSameTone) {
   auto keys = asciiKeys("3u3");
   auto endState = handleKeySequence(keys);
   auto inputtingState = dynamic_cast<InputStates::Inputting*>(endState.get());
   ASSERT_TRUE(inputtingState != nullptr);
-  ASSERT_EQ(inputtingState->composingBuffer, "以");
-  ASSERT_EQ(inputtingState->cursorIndex, strlen("以"));
+  ASSERT_EQ(inputtingState->composingBuffer, "以ˇ");
+  ASSERT_EQ(inputtingState->cursorIndex, strlen("以ˇ"));
 }
 
 TEST_F(KeyHandlerTest, ToneMarkThenNonToneComponentOnlyComposesWithSpace) {
   auto keys = asciiKeys("3u ");
   auto endState = handleKeySequence(keys);
-  auto inputtingState = dynamic_cast<InputStates::Inputting*>(endState.get());
+  auto inputtingState =
```

Comment:
Rename test case to `ToneMarkThenNonToneComponentResultingInCompositionCase4`, then add a comment that the last Space key resulting in entering the `ChoosingCandidate` state?

Reply:
Done.